salt/cloud/clouds/azurearm.py

```diff
@@ -1164,6 +1167,12 @@ def request_instance(vm_):
         if vm_['image'].startswith('http'):
             source_image = VirtualHardDisk(vm_['image'])
             img_ref = None
+            if not os_type:
+                log.debug("os_type not specified")
+                if win_installer:
+                    os_type = 'Windows'
+                else:
+                    os_type = 'Linux'
```

Comment:
This is going to cause an issue, can we put this statement back where it was before?